api/src/pcapi/core/external/compliance.py

```diff
@@ -0,0 +1,130 @@
+                },
+                json={"audience": client_id, "includeEmail": "true"},
+            )
+            id_token = id_token_response.json()["token"]
+        except requests.exceptions.InvalidHeader:
+            # The auth_token has probably not been changed to a correct value
+            logger.info("Check the value of your auth_token")
+            data_score = None
+            return
+        except Exception as exc:  # pylint: disable=broad-except
+            logger.info("Could not get id_token", extra={"exc": exc})
+            data_score = None
+            return
+
+    else:
+        id_token = auth_api.get_id_token_from_google(client_id)
+
+    data = payload.to_dict()
+    try:
+        api_response = requests.post(
+            "https://compliance.passculture.team/latest/model/compliance/scoring",
+            headers={
+                "Authorization": f"Bearer {id_token}",
+                "accept": "application/json",
+                "Content-Type": "application/json",
+            },
+            json=data,
+        )
+        data_score = api_response.json()["probability_validated"]
+
+    except Exception as exc:  # pylint: disable=broad-except
+        logger.exception("Connection to API Compliance failed", extra={"exc": exc})
+        data_score = None
+
+    if data_score:
+        if offer.extraData is None:
+            offer.extraData = {}
+        offer.extraData["complianceScore"] = data_score
+    db.session.add(offer)
+    db.session.commit()
+
+
+def _get_payload_for_compliance_api(offer: offers_models.Offer) -> GetComplianceScoreRequest:
+    extra_data = offer.extraData or {}
+    rayon = extra_data.get("rayon")
+    macro_rayon = (
+        (
+            offers_models.BookMacroSection.query.filter(sa.func.lower(offers_models.BookMacroSection.section) == rayon)
+            .with_entities(offers_models.BookMacroSection.macroSection)
+            .one_or_none()
+        )
+        if rayon
+        else None
+    )
+
+    offer_type_label = None
+    if show_type := extra_data.get("showType"):
+        offer_type_label = show_types.SHOW_TYPES_LABEL_BY_CODE[int(show_type)]
+    elif music_type := extra_data.get("musicType"):
+        offer_type_label = music_types.MUSIC_TYPES_LABEL_BY_CODE[int(music_type)]
+
+    offer_sub_type_label = None
+    if show_sub_type := extra_data.get("showSubType"):
+        offer_sub_type_label = show_types.SHOW_SUB_TYPES_LABEL_BY_CODE[int(show_sub_type)]
+    elif music_sub_type := extra_data.get("musicSubType"):
+        offer_sub_type_label = music_types.MUSIC_SUB_TYPES_LABEL_BY_CODE[int(music_sub_type)]
+
+    payload = GetComplianceScoreRequest(
+        offer_id=str(offer.id),
+        offer_name=offer.name,
+        offer_description=offer.description,
+        offer_subcategoryid=offer.subcategoryId,
+        rayon=rayon,
+        macro_rayon=macro_rayon,
+        stock_price=str(offer.max_price),
```

Comment:
`max_price` fait `max(stock.price for stock in self.stocks if not stock.isSoftDeleted)`, donc va avoir besoin de faire une requête supplémentaire pour le stock. On pourrait probablement `joinedload` le stock (juste les colonnes dont on a besoin) au début de `make_update_offer_compliance_score` afin de tout récupérer d'un coup.
Et vérifier ce nombre de requêtes dans le test auto.